packages/bazel-bot/docker-image/install-credentials.sh

```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script uses the github app secret to install credentials in the GOOGLEAPIS_GEN
+# repo and the 'gh' command line tool.
+
+# According to https://docs.github.com/en/developers/apps/authenticating-with-github-apps#authenticating-as-a-github-app
+JWT=$(jwt encode --secret "$GITHUB_APP_SECRET" --iss "$GITHUB_APP_ID" --exp "+10 min" --alg RS256)
+
+# According to https://docs.github.com/en/developers/apps/authenticating-with-github-apps#authenticating-as-an-installation
+GITHUB_TOKEN=$(curl -X POST \
+    -H "Authorization: Bearer $JWT" \
+    -H "Accept: application/vnd.github.v3+json" \
+    https://api.github.com/app/installations/$GITHUB_APP_INSTALLATION_ID/access_tokens \
+    | jq -r .token)
+
+# According to https://cli.github.com/manual/gh_auth_login
+echo "$GITHUB_TOKEN" | gh auth login --with-token
+
+# According to https://docs.github.com/en/developers/apps/authenticating-with-github-apps#http-based-git-access-by-an-installation
+git -C "$GOOGLEAPIS_GEN" remote set-url origin \
```

Comment:
I used the credentials helper to setup the authentication, I'm wondering if that makes it less likely that a URL shows up in logs with the token?

https://github.com/googleapis/repo-automation-bots/blob/master/packages/owl-bot-post-processor/cloud-build/update-pr.yaml#L51

Reply:
I tried that too.  However, I got a permission denied issue.  I think I'm using it in a different context, maybe because I clone with https and later switch the url to ssl.

Reply:
👍 we should just make sure that the Cloud Build logs are anywhere publicly accessible, and that the token doesn't show up in the logs.